src/pubsub_test/pubsub_sync_client.clj

```diff
@@ -0,0 +1,182 @@
+      .build))
+
+(defn send-mod-acks
+  [ack-deadline-seconds-increament to-mod-ack subscriber sub-name]
+  (try
+    (when-not (empty? @to-mod-ack)
+      (log/info "Sending mod-acks: " @to-mod-ack)
+      (-> subscriber
+          (.modifyAckDeadlineCallable)
+          (.call (mod-ack-request ack-deadline-seconds-increament sub-name @to-mod-ack))))
+    (catch Exception e
+      (log/error e))))
+
+(defn ack-request
+  [sub-name ack-ids]
+  (-> (AcknowledgeRequest/newBuilder)
+      (.setSubscription sub-name)
+      (.addAllAckIds (ArrayList. ack-ids))
+      .build))
+
+(defn send-acks
+  [to-ack to-mod-ack subscriber sub-name]
+  (try
+    (let [local-to-acks @to-ack]
+      (when-not (empty? local-to-acks)
+        (log/info "Sending acks: " local-to-acks)
+        (-> subscriber
+            (.acknowledgeCallable)
+            (.call (ack-request sub-name local-to-acks)))
+        (swap! to-ack #(apply disj % local-to-acks))
+        (swap! to-mod-ack #(apply disj % local-to-acks))))
+    (catch Exception e
+      (log/error e))))
+
+(defn process-messages
+  [message-fn worker-pool to-ack to-mod-ack subscriber subname num-threads]
+  (try
+    (when (> num-threads (count @to-mod-ack))
+      (let [pull-response (-> subscriber
+                              (.pullCallable)
+                              (.call (pull-request subname (- num-threads (count @to-mod-ack)))))]
+        (doseq [message (.getReceivedMessagesList pull-response)]
+          (let [ack-id (.getAckId message)
+                future (.submit worker-pool (partial message-fn message))]
+            (swap! to-mod-ack conj ack-id)
+            (Futures/addCallback future (proxy [FutureCallback] []
+                                          (onSuccess [message-id]
+                                            (swap! to-ack conj ack-id))
+                                          (onFailure [thrown]
+                                            (swap! to-mod-ack disj ack-id))))))))
+    (catch Exception e
+      (log/error e))))
+
+(defn schedule-fn
+  [project-id subscription-id pull-interval-milliseconds num-threads max-rpc-seconds]
+  (fn [message-fn]
+    (let [subscriber (GrpcSubscriberStub/create (client-settings max-rpc-seconds))
+          send-ack-period-seconds (int 1)
+          max-rpc-seconds (int 20)
+          ack-deadline-seconds-increament (int 40)
+          send-mod-ack-period-seconds (int 30)
+          worker-pool (MoreExecutors/listeningDecorator (Executors/newFixedThreadPool num-threads))
+          service-pool (Executors/newScheduledThreadPool 4)
+          sub-name (str (SubscriptionName/of project-id subscription-id))
+          to-mod-ack (atom #{})
+          to-ack (atom #{})
+          inital-delay 0]
+      (.scheduleWithFixedDelay
+       service-pool
+       #(process-messages message-fn worker-pool to-ack to-mod-ack subscriber sub-name num-threads)
+       inital-delay
+       pull-interval-milliseconds
+       TimeUnit/MILLISECONDS)
```

Comment:
Changed this to milliseconds as a quick way round the buffer issue. This should be fine for now, there are better methods but we just need to check it work's first then make it faster